### ISSUE TITLE
Remove Intellij IDEA specific files from project

### ISSUE THREAD

**Issue by vbauer** [OfficeDev/ews-java-api#235]
I think it will be better to:
- remove ".idea" folder and ".iml"-file
- update .gitignore file

For Maven projects is completely unnecessary. Code style configuration (and other IDEA-specific files) could be placed in the some new folder like "misc" or "misc/IDEA".

**Comment by serious6** [comment]
problem is with keeping the files the formatter will be used automatically since it is project default. If we'll have this seperated the developer has to set the formatter on it's own. 
This can be somehow dangerous since we don't have an automated checkstyle configuration (before this is possible we need all violations to be fixed) to see if the PR matches our formatting criterias. 
I have seen an intelliJ maven plugin but this one is outdated but maybe you have some other ideas?

**Comment by vbauer** [comment]
Just to share my experience:

In all my projects I use [maven-checkstyle-plugin](http://maven.apache.org/plugins/maven-checkstyle-plugin) without any IDE formatters, so I can check my code and code of other people on CI server (instead of manual code review). Automatic analysis tools looks better in this case.

**Comment by vbauer** [comment]
I want to add some other thoughts about this task. In the current situation files `.idea/codeStyleSettings.xml` and `.idea/misc.xml` are tracked by Git and Intellij Idea:

Idea IDE usually wins the war and changes this files in someway. After that Git finds the untracked changes and there is a problem: we have a dirty working branch (unstaged changes), so we have to do something like in the command-line:

* `git checkout .idea/codeStyleSettings.xml` and `git checkout .idea/misc.xml`
* or just `git checkout .idea`

As result, it is uncomfortable to work with Git branches in the terminal.

**Comment by evpaassen** [comment]
Currently, we're following the [guidelines from the JetBrains docs](https://intellij-support.jetbrains.com/entries/23393067-How-to-manage-projects-under-Version-Control-Systems).

It's true that IntelliJ sometimes alters some of the files checked in, which is annoying. But I think  `.idea/codeStyleSettings.xml` is not one of them, unless you make changes to the code style settings yourself.

Using a tool (Maven plugin?) to generate a project config could be a solution. I know such a plugin exists for Gradle, but I don't know about Maven.

**Comment by vbauer** [comment]
In my environment `.idea/codeStyleSettings.xml` and `.idea/misc.xml` are changed each time when I work with ews-java-api. I think it could happens, because we have different versions of Intellij IDEA, and IDE converts some parameters (or add some new defaults).

Btw, Maven team also uses codestyles for Intellij Idea as separate configuration: http://maven.apache.org/developers/committer-environment.html

```
Download maven-idea-codestyle.xml and copy it to ~/.IntelliJIDEA/config/codestyles then restart IDEA. On Windows, try C:\Documents and Settings\<username>\.IntelliJIDEA\config\codestyles

After this, restart IDEA and open the settings to select the new code style.
```

**Comment by serious6** [comment]
maybe these changes occur because you dont just open the project on intelliJ? If you create or import a project this might happen.

**Comment by vbauer** [comment]
I've just cloned git repo again and done a clean test: "File -> Open -> ... -> ews-java-api -> pom.xml" That's all.

Results:

New lines in `.idea/codeStyleSettings.xml`:
```xml
        <codeStyleSettings language="Groovy">
          <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
          <option name="PARENT_SETTINGS_INSTALLED" value="true" />
        </codeStyleSettings>
        <codeStyleSettings language="JavaScript">
          <option name="WRAP_COMMENTS" value="true" />
          <option name="PARENT_SETTINGS_INSTALLED" value="true" />
        </codeStyleSettings>
        <codeStyleSettings language="TypeScript">
          <option name="WRAP_COMMENTS" value="true" />
          <option name="PARENT_SETTINGS_INSTALLED" value="true" />
        </codeStyleSettings>
```

New line in `.idea/misc.xml`:
```xml
<component name="IdProvider" IDEtalkID="49369D9F2811625BADF7D0C400625C75" />
```

**Comment by evpaassen** [comment]
I guess we'll have to look at a decent solution using a build tool plugin which can generate the project config...

**Comment by serious6** [comment]
Is this a guess or do you have something in mind?

**Comment by evpaassen** [comment]
I though about it and I think we could just add the `.idea` folder and `ews-java-api.iml` to the `.gitignore` file. This will keep the files where they are, but updates to the files will be ignored by git. If we ever need to update these files again, they can still be added to the git index using `git add -f <filename>`.

**Comment by vbauer** [comment]
Please increase priority of this ticket, it is really uncomfortable to work on project.

**Comment by evpaassen** [comment]
> I thought about it and I think we could just add the `.idea` folder and `ews-java-api.iml` to the `.gitignore` file. This will keep the files where they are, but updates to the files will be ignored by git. If we ever need to update these files again, they can still be added to the git index using `git add -f <filename>`.

Turns out this won't work. If files have been added already, git keeps tracking changes in it, even if they are in the `.gitignore` file.

**Comment by vbauer** [comment]
> Turns out this won't work. If files have been added already, git keeps tracking changes to it, even if the file is in the .gitignore file.

Yes, I've just faced with it here https://github.com/OfficeDev/ews-java-api/issues/283

**Comment by xvik** [comment]
+1 for removing

**Comment by pathikrit** [comment]
+1 - please remove IDE specific files.
Here's a canonical gitignore for Jetbrains which ignores anything under .idea/
https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore

**Comment by vbauer** [comment]
`.gitignore` file could be also generated using [gitignore.io](http://gitignore.io) to prevent a lot of unnecessary files generated by Eclipse, Windows, OS X, Netbeans, IDEA, etc..

**Comment by dennis-f** [comment]
using .idea/* in gitignore works best for me

**Comment by trules** [comment]
I had this issue and manages to resolve using [the method described here](http://stackoverflow.com/questions/1274057/how-to-make-git-forget-about-a-file-that-was-tracked-but-is-now-in-gitignore)
